beacon_node/beacon_chain/src/historical_blocks.rs

```diff
@@ -135,20 +135,20 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
 
             prev_block_slot = block.slot();
             expected_block_root = block.message().parent_root();
+            signed_blocks.push(block);
 
             // If we've reached genesis, add the genesis block root to the batch for all slots
             // between 0 and the first block slot, and set the anchor slot to 0 to indicate
             // completion.
             if expected_block_root == self.genesis_block_root {
                 let genesis_slot = self.spec.genesis_slot;
-                for slot in genesis_slot.as_usize()..block.slot().as_usize() {
+                for slot in genesis_slot.as_usize()..prev_block_slot.as_usize() {
```

Comment:
`prev_block_slot` case is already covered in the previous iteration in lines

https://github.com/sigp/lighthouse/blob/0aa5ed2f7c4afe321591749fa3a12ba4baef903b/beacon_node/beacon_chain/src/historical_blocks.rs#L131-L134